### PR TITLE
Adds org naming step in dedicated setup flow

### PR DESCRIPTION
**Pull request by selfcontained**
## Description

This adds the org naming (and creation) step to the dedicated setup flow. I also stubbed out the sso config step, and final completed view so that they're ready to build out in subsequent PRs.

image

## Related Issue(s)

Fixes WEB-244

## How to test

* Since the setup flow isn't complete end to end yet, the easiest way to test this it to start a workspace using this PR/branch, and point gitpod-staging at your workspace dashboard assets w/ the `x-frontend-dev-url` header.
* Load up gitpod staging and set a query param that will initiate the flow.
* https://gitpod-staging.com/workspaces?dedicated-setup=force
* You should see the getting started step, and after proceeding, you can name your org. 
* After that completes you should land on a placeholder for the sso setup page in the flow.

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - bmh-dedicaad8deebe14</li>
	<li><b>🔗 URL</b> - <a href="https://bmh-dedicaad8deebe14.preview.gitpod-dev.com/workspaces" target="_blank">bmh-dedicaad8deebe14.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] analytics=segment
- [x] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold

### CODE REVIEW

File: components/dashboard/src/dedicated-setup/OrgNamingStep.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+type Props = {
+    onComplete: () => void;
+};
+export const OrgNamingStep: FC<Props> = ({ onComplete }) => {
+    // TODO: if there's already an org created, set initial value to current org, or we could skip this step
+    const [orgName, setOrgName] = useState("");
+    const createOrg = useCreateOrgMutation();
+
+    const handleContinue = useCallback(async () => {
+        createOrg.mutate({ name: orgName }, { onSuccess: () => onComplete() });
```
Comment (svenefftinge) [inline components/dashboard/src/dedicated-setup/OrgNamingStep.tsx:25]: unnecessary arrow function? `onSuccess: onComplete`

File: components/dashboard/src/data/organizations/create-org-mutation.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { Organization } from "@gitpod/gitpod-protocol";
+import { useMutation } from "@tanstack/react-query";
+import { useOrganizationsInvalidator } from "./orgs-query";
+import { publicApiTeamToProtocol, teamsService } from "../../service/public-api";
+
+type UpdateOrgArgs = Pick<Organization, "name" | "slug">;
```
Comment (svenefftinge) [inline components/dashboard/src/data/organizations/create-org-mutation.ts:12]: The name should be CreateOrgArgs, but we only need a name, so I think we can just use `string`

**Review by svenefftinge — APPROVED** [review]

File: components/dashboard/src/dedicated-setup/DedicatedSetup.tsx [hunk 3]
```diff
@@ -4,50 +4,50 @@
...
 type Props = {
     onComplete?: () => void;
 };
 const DedicatedSetup: FC<Props> = ({ onComplete }) => {
-    const handleGetStarted = useCallback(() => {
-        console.log("getting started");
-    }, []);
+    const { dropConfetti } = useConfetti();
+    // TODO: try and determine what step we should start on based on current state, i.e. org/sso already exists
+    const [step, setStep] = useState<StepsValue>(STEPS.GETTING_STARTED);
```
[the remark this answers is not in the document]
Comment (selfcontained) [reply components/dashboard/src/dedicated-setup/DedicatedSetup.tsx:29]: Agree, there's a new api method we're going to add that will return if the installation needs setup, and give us details on what steps are needed. This logic will rely on that once it's ready.

File: components/dashboard/src/dedicated-setup/OrgNamingStep.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+import { Button } from "../components/Button";
+import { useOnBlurError } from "../hooks/use-onblur-error";
+import { useCreateOrgMutation } from "../data/organizations/create-org-mutation";
+import Alert from "../components/Alert";
+
+type Props = {
+    onComplete: () => void;
+};
+export const OrgNamingStep: FC<Props> = ({ onComplete }) => {
+    // TODO: if there's already an org created, set initial value to current org, or we could skip this step
```
[the remark this answers is not in the document]
Comment (selfcontained) [reply components/dashboard/src/dedicated-setup/OrgNamingStep.tsx:20]: Yah could be that works best. I'm holding off until we have a new api that gives us some of these onboarding details to see how that plugs in.

**Pull request merged**